platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectContainerExtensions.kt

```diff
@@ -254,15 +255,23 @@ internal constructor(
      * @see [NamedDomainObjectProvider.configure]
      */
     operator fun String.invoke(configuration: T.() -> Unit): NamedDomainObjectProvider<T> =
-        this().apply { configure(configuration) }
+        named(this).apply { configure(configuration) }
 
     /**
      * Locates an object by name, without triggering its creation or configuration, failing if there is no such object.
      *
      * @see [NamedDomainObjectContainer.named]
      */
-    operator fun String.invoke(): NamedDomainObjectProvider<T> =
-        delegate.named(this)
+    @Deprecated("Use named(String) instead.", ReplaceWith("named(\"...\")"))
```

Comment:
❌ We can fully automate replacing this properly with different syntax.
```suggestion
    @Deprecated("Use `named(String)` instead.", ReplaceWith("named(this)"))
```
Though I would verify that this actually works, I believe this is correct according to the [documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-replace-with/expression.html). 

💭 I also changed the message to use backticks for clarity.